betterproto/tests/test_get_ref_type.py

```diff
@@ -77,6 +79,227 @@ def test_importing_google_wrappers_unwraps_them(google_type: str, expected_name:
+    assert name == "deeply_nested_child.Message"
+
+
+def test_import_root_sibling():
+    imports = set()
+    name = get_ref_type(package="", imports=imports, source_type="Message")
+
+    assert imports == set()
+    assert name == '"Message"'
+
+
+def test_import_nested_siblings():
+    imports = set()
+    name = get_ref_type(package="foo", imports=imports, source_type="foo.Message")
+
+    assert name == '"Message"'
+
+
+def test_import_deeply_nested_siblings():
+    imports = set()
+    name = get_ref_type(
+        package="foo.bar", imports=imports, source_type="foo.bar.Message"
+    )
+
+    assert name == '"Message"'
+
+
+def test_import_parent_package_from_child():
+    imports = set()
+    name = get_ref_type(
+        package="package.child", imports=imports, source_type="package.Message"
+    )
+
+    assert imports == {"from ... import package as __package__"}
+    assert name == "__package__.Message"
+
+
+def test_import_parent_package_from_deeply_nested_child():
+    imports = set()
+    name = get_ref_type(
+        package="package.deeply.nested.child",
+        imports=imports,
+        source_type="package.deeply.nested.Message",
+    )
+
+    assert imports == {"from ... import nested as __nested__"}
+    assert name == "__nested__.Message"
+
+
+def test_import_ancestor_package_from_nested_child():
+    imports = set()
+    name = get_ref_type(
+        package="package.ancestor.nested.child",
+        imports=imports,
+        source_type="package.ancestor.Message",
+    )
+
+    assert imports == {"from .... import ancestor as ___ancestor__"}
+    assert name == "___ancestor__.Message"
+
+
+def test_import_root_package_from_child():
+    imports = set()
+    name = get_ref_type(package="package.child", imports=imports, source_type="Message")
+
+    assert imports == {"from ... import Message"}
+    assert name == "Message"
+
+
+def test_import_root_package_from_deeply_nested_child():
+    imports = set()
+    name = get_ref_type(
+        package="package.deeply.nested.child", imports=imports, source_type="Message"
+    )
+
+    assert imports == {"from ..... import Message"}
+    assert name == "Message"
+
+
+def test_import_unrelated_package():
+    imports = set()
+    name = get_ref_type(package="a", imports=imports, source_type="p.Message")
+
+    assert imports == {"from .. import p as _p__"}
+    assert name == "_p__.Message"
+
+
+def test_import_unrelated_nested_package():
+    imports = set()
+    name = get_ref_type(package="a.b", imports=imports, source_type="p.q.Message")
+
+    assert imports == {"from ...p import q as __p_q__"}
+    assert name == "__p_q__.Message"
+
+
+def test_import_unrelated_deeply_nested_package():
+    imports = set()
+    name = get_ref_type(
+        package="a.b.c.d", imports=imports, source_type="p.q.r.s.Message"
+    )
+
+    assert imports == {"from .....p.q.r import s as ____p_q_r_s__"}
+    assert name == "____p_q_r_s__.Message"
```

Comment:
😧

Reply:
Yep 😓 100% ugly, but 100% reliable.

Won't ever create name conflicts. Hopefully we can get rid of it in some more advanced stage of betterproto!